typescript/packages/plugins/balmy/src/balmy.plugin.ts

```diff
@@ -0,0 +1,692 @@
+        publicRPCs: [
+            "https://goerli.base.org",
+            "https://base-goerli.public.blastapi.io",
+            "https://base-goerli.blockpi.network/v1/rpc/public",
+            "https://endpoints.omniatech.io/v1/base/goerli/public",
+            "https://base-goerli.gateway.tenderly.co",
+        ],
+        explorer: "https://goerli.basescan.org/",
+        testnet: true,
+    },
+};
+
+const SUPPORTED_CHAIN_IDS = [
+    1, // Ethereum
+    10, // Optimism
+    42161, // Arbitrum
+    42170, // Arbitrum Nova
+    137, // Polygon
+    56, // BNB Chain
+    8453, // Base
+    250, // Fantom
+    42220, // Celo
+    1088, // Metis
+    43114, // Avalanche
+    128, // Heco
+    66, // OKC
+    1285, // Moonriver
+    1284, // Moonbeam
+    122, // Fuse
+    106, // Velas
+    100, // Gnosis
+    25, // Cronos
+    288, // Boba
+    58, // Ontology
+    8217, // Kaia
+    1313161554, // Aurora
+    592, // Astar
+    1666600000, // Harmony
+    199, // BitTorrent
+    42262, // Oasis
+    204, // opBNB
+    7700, // Canto
+    9001, // Evmos
+    30, // Rootstock
+    1101, // Polygon zkEVM
+    2222, // Kava
+    59144, // Linea
+    34443, // Mode
+    81457, // Blast
+    534352, // Scroll
+    5000, // Mantle
+    // Testnets
+    5, // Goerli
+    11155111, // Sepolia
+    80001, // Mumbai
+    84531, // Base Goerli
+];
+
+const DEFAULT_CONFIG = {
+    quotes: {
+        sourceList: {
+            type: "local" as const,
+        },
+        defaultConfig: {
+            global: {
+                slippagePercentage: 0.5,
+                txValidFor: "180s",
+                disableValidation: true,
+            },
+            custom: {
+                bebop: { enabled: false, apiKey: "" },
+                "0x": { enabled: true, apiKey: "", baseUrl: "https://api.0x.org" },
+                "li-fi": { apiKey: "" },
+                paraswap: { enabled: false, sourceAllowlist: [], sourceDenylist: [] },
+                "1inch": { enabled: false, customUrl: "", sourceAllowlist: [] },
+            },
+        },
+    },
+    provider: {
+        source: {
+            type: "http" as const,
+            url: "", // Will be set dynamically
+            supportedChains: SUPPORTED_CHAIN_IDS,
+        },
+        defaultConfig: {
+            chainId: 1, // Will be set dynamically
+            rpcUrl: "", // Will be set dynamically
+        },
+    },
+};
+
+export class BalmyPlugin extends PluginBase {
+    private balmyService: BalmyService;
+
+    constructor() {
+        const service = new BalmyService(DEFAULT_CONFIG);
+        super("balmy", [service]);
+        this.balmyService = service;
+    }
+
+    supportsChain = (chain: Chain) => {
+        if (chain.type !== "evm" || !SUPPORTED_CHAIN_IDS.includes(chain.id)) {
+            return false;
+        }
+
+        // Update config with chain-specific RPC URL
+        const rpcUrl = this.getRpcUrlForChain(chain.id);
```

Comment:
This is def not the place for this. Also not needed, you are defaulting to public RPCs which the SDK already does.